include/na/azac/VMPlacer.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "ir/Definitions.hpp"
+#include "na/azac/Architecture.hpp"
+#include "na/azac/Types.hpp"
+
+#include <cstddef>
+#include <functional>
+#include <nlohmann/json_fwd.hpp>
+#include <optional>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+namespace na::azac {
+/// class to find a qubit layout based on vertex matching of a weighted
+/// bipartite graph
+class VMPlacer {
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatching1_Test;
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatching2_Test;
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatchingExceptions_Test;
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatchingEmpty_Test;
+
+  std::reference_wrapper<const Architecture> architecture_;
+  /// If true, during the initial placement the atoms are placed starting in the
+  /// last row instead of the first row in the first SLM
+  bool reverseInitialPlacement_ = false;
+
+  /// this flag indicates whether the  placement should use a window when
+  /// selecting potential free sites
```

Comment:
```suggestion
  /// this flag indicates whether the placement should use a window when
  /// selecting potential free sites
```
Double whitespace here. 
Also: it is unclear from the documentation up until this point what a "window" is.